### DISCUSSION TITLE
[bug] can't connect to tauri.localhost

### DISCUSSION THREAD

**Discussion by danitor1** [tauri-apps/tauri#4052] [General]
Hi guys. I copy and paste a question made in other web because I have exactly the same problem and I need help:

Using angular when opening a router link on the build version always gives a
"This site can’t be reached; tauri.localhost refused to connect";
This does not happen on the dev version due to the fact that the dev version is not running on tauri.localhost but on http://localhost:4200/.

In order for a route link to work on angular every link must lead back to index.html. I assume this works on the dev version since it receives information from the local webpack server. However, on the build version when a new window opens there is no information being sent so the routes always fail to complete. This happens on something as simple as a or even left click open link on a new window.

I don't understand what the issue could be or if there is a way to resolve it so that on the installed version these new windows actually complete.

Can you explain the solution? Thanks a lot.

**Comment by danitor1** [comment]
THE ANSWER:
This happens when you install Tauri on an Angular project. You have to change the "distPath" from "../dist" to "../dist/nameOfYourProject" and the .exe will work perfectly.

**Reply by mota57** [reply]
I have the same issue on next.js but fine.. I will try that.

**Reply by mota57** [reply]
no, issue persist.

**Reply by ZeeshanAhmadKhalil** [reply]
same issue for me looks like its not picking up the out/dist folder

**Reply by ZeeshanAhmadKhalil** [reply]
In my case issue was on the `next.js` side. I was not creating the `pages/index.html` file which becomes the entry file `index.html` after `next build && next export`.

Due to this reason was getting a white screen. After adding `pages/index.html` and redirecting to my main app, the issue got resolved.

**Reply by Tenkaklet** [reply]
Thanks mate! This made rush outta bed a continue coding! 🥇

**Reply by lucasmolinari** [reply]
>This happens when you install Tauri on an Angular project. You have to change the "distPath" from "../dist" to "../dist/nameOfYourProject" and the .exe will work perfectly.

You actually need to point to the ```browser``` folder too if that is the case. The full path should be: ```../dist/[ProjectName]/browser```

**Reply by aidanwardman** [reply]
This did not work for me, /out folder only contains [_next,404.html,next.svg,vercel.svg] no index.html file ever generated

**Comment by NomadicDeveloper22** [comment]


**Reply by FabianLars** [reply]
Are you using the static-adapter? It's required to make sveltekit work in Tauri (and similar GUI frameworks).

We also give a short intro on that in our guide https://tauri.app/v1/guides/getting-started/setup/sveltekit#sveltekit-in-ssg-mode

**Reply by FabianLars** [reply]
> I'm gonna put my plans for using Tauri on hiatus for now. But am I curious as to why is works flawlessly with run dev while building an executable leads to bad results

Because dev mode uses sveltekit's dev server while build mode doesn't have a server so only code that's _fully_ SPA/SSG compatible can work.
That's one of the reasons i rarely recommend sveltekit for tauri apps (or similar gui frameworks). The difference between dev and build is just too big imho and it's often not exactly obvious what won't work in build mode.

This is the new link for the 404 page btw: https://tauri.app/v1/guides/building/sidecar

**Comment by well-hello-there** [comment]
Anyone else having this issue with react native / expo?

**Comment by Yu15693** [comment]
It seems that Tauri only looks for `index.html` in the root path of `distDir`.
Therefore, check your dist directory to see whether the `index.html` is in the correct location.

**Reply by aidanwardman** [reply]
Nextjs 14.2.10 , Tauri 1.6.0 , NPM v10.8.1 , Node v20.16.0 , no index.html file generated in /out (distDir folder), next.config.mjs set output:'export' , on successful build, running app gives cannot connect to localhost.

**Comment by th3oth3rjak3** [comment]
I'm also having this problem when generating a brand new application with a vue template. It says "cannot connect to localhost:1420" but the other steps above did not work for me. Anyone else able to fix this with vue?

**Reply by aidanwardman** [reply]
Same issue here Nextjs 14.2.10 , Tauri 1.6.0 , NPM v10.8.1 , Node v20.16.0 , no index.html file generated in /out (distDir folder), next.config.mjs set output:'export' , on successful build, running app gives cannot connect to localhost.
